### PR TITLE
[#noissue] Fixed bug with fields validation

### PR DESCRIPTION
**Pull request by Evengining**
Now the filter field is checked for emptiness. The request field will also be checked (in the future, if all goes well). I also refactor csearch, there is also a check, filter and query.

### CODE REVIEW

**Comment by Evengining** [comment]
Also @ppressives look at it https://github.com/Evengining/codesearch if you want.

File: web-server/app/codesearch/web/controllers/SearchController.scala [hunk 1]
```diff
@@ -36,16 +36,21 @@ trait SearchController[V <: DefaultTable] { self: InjectedController =>
...
              sources: String,
              page: String): Action[AnyContent] =
     Action.async { implicit request =>
-      val request = SearchRequest.applyRaw(query, filter, caseInsensitive, spaceInsenstive, precise, sources, page)
-      val callURI =
-        s"/$lang/search?query=$query?filter=$filter&caseInsensitive=$caseInsensitive&spaceInsensitive=$spaceInsenstive&precise=$precise&sources=$sources"
+      val request =
+        SearchRequest.applyRaw(query, filter, caseInsensitive, spaceInsenstive, precise, sources, page)
+
+      val callURI = filter match {
```
Comment (kamilongus) [inline web-server/app/codesearch/web/controllers/SearchController.scala]: You can make the `callURI ` function a member of the `SearchRequest` case class

File: web-server/conf/routes [hunk 2]
```diff
@@ -3,19 +3,19 @@ GET        /                                 codesearch.web.controllers.Applicat
 GET        /*path/                           codesearch.web.controllers.Application.untrail(path: String)
 
 GET        /haskell                          codesearch.web.controllers.HackageSearcher.index
-GET        /haskell/search                   codesearch.web.controllers.HackageSearcher.search(query: String ?= "", filter: String ?= "", insensitive: String ?= "", space: String ?= "", precise: String ?= "", sources: String ?= "on", page: String ?= "1")
+GET        /haskell/search                   codesearch.web.controllers.HackageSearcher.search(query: String ?= "", filter: Option[String], insensitive: String ?= "", space: String ?= "", precise: String ?= "", sources: String ?= "on", page: String ?= "1")
```
Comment (kamilongus) [inline web-server/conf/routes]: Why not make all the parameters optional?

**Comment by neongreen** [comment]
Assigning @kamilongus 'cause there's nobody else.

File: web-server/app/codesearch/web/controllers/SearchController.scala (hunk 1 above)
Comment (Evengining) [reply web-server/app/codesearch/web/controllers/SearchController.scala]: I did, but I'm not sure if it is https://github.com/aelve/codesearch/pull/192/commits/6e4c56b5c3046930e607876f7887872685db6b12

File: web-server/conf/routes (hunk 2 above)
Comment (Evengining) [reply web-server/conf/routes]: What for? Do we have all the fields, either true or false, and the query doesn't exactly have to be optional?

File: web-server/app/codesearch/web/controllers/SearchController.scala (hunk 1 above)
Comment (Evengining) [reply web-server/app/codesearch/web/controllers/SearchController.scala]: @kamilongus I think that it was not done in the best way, you just said it, but I think you meant something else.

File: core/src/main/scala/codesearch/core/search/SearchRequest.scala [hunk 3]
```diff
@@ -17,7 +17,22 @@ case class SearchRequest(
     preciseMatch: Boolean,
     sourcesOnly: Boolean,
     page: Int
-)
+) {
+  def callURI(
```
Comment (kamilongus) [inline core/src/main/scala/codesearch/core/search/SearchRequest.scala]: Redundant parameters.

File: core/src/main/scala/codesearch/core/search/SearchRequest.scala [hunk 4]
```diff
@@ -16,42 +16,39 @@ case class SearchRequest(
...
-    isEnabled(insensitive),
-    isEnabled(spaceInsensitive),
-    isEnabled(preciseMatch),
-    isEnabled(sourcesOnly),
-    page.toInt
-  )
+  ): SearchRequest = {
+    val callURI: String = filter match {
+      case Some(filter) =>
+        s"/$lang/search?query=$query&filter=$filter&insensitive=$insensitive&space=$spaceInsensitive&precise=$preciseMatch&sources=$sourcesOnly"
```
Comment (kamilongus) [inline core/src/main/scala/codesearch/core/search/SearchRequest.scala]: You have lost the `page` parameter

File: web-server/app/views/searchBox.scala.html [hunk 5]
```diff
@@ -23,63 +22,50 @@
...
+    this.filter = document.getElementById("filter").value
+    this.inpCheck = document.querySelectorAll('input[type="checkbox"]')
 
-                        const query = inputsValidator(fieldForm("query"))
-                        const filter = inputsValidator(fieldForm("filter"))
-                        const insensitive = checkBoxValidator(fieldForm("insensitive"))
-                        const space = checkBoxValidator(fieldForm("space"))
-                        const precise = checkBoxValidator(fieldForm("precise"))
-                        const sources = inputsValidator(fieldForm("sources"))
+    if (this.query != '') {
```
Comment (kamilongus) [inline web-server/app/views/searchBox.scala.html]: Call the 'trim' function for `this.query`.

File: web-server/app/views/searchBox.scala.html [hunk 6]
```diff
@@ -65,7 +66,10 @@
 
     } else {
         document.getElementById('query').classList.add('warn')
```
Comment (kamilongus) [inline web-server/app/views/searchBox.scala.html]: It nowhere removed after

File: web-server/app/views/searchBox.scala.html [hunk 7]
```diff
@@ -65,7 +66,10 @@
...
-        document.getElementById('input-group').insertBefore(div, document.getElementById('input-group').children[1])
+        if (!counterClicks) {
+            document.getElementById('input-group').insertBefore(div, document.getElementById('input-group').children[1])
+            counterClicks++
```
Comment (kamilongus) [inline web-server/app/views/searchBox.scala.html]: Why not a boolean variable?

File: web-server/app/views/searchBox.scala.html [hunk 8]
```diff
@@ -28,48 +28,67 @@
...
-function FormValidate() {
+        function focusQueryInput() {
 
-    const sendArr = []
+            dQuery.addEventListener('input', function(e){
 
-    const div = document.createElement('div')
-        div.className = "warning"
-        div.innerHTML = "Заполните это поле!"
+                if(dQuery.value.length >= 1) {
```
[the remark this answers is not in the document]
Comment (kamilongus) [reply web-server/app/views/searchBox.scala.html]: > Call trim, f.e
> `dQuery.value.trim().length`

+++

**Comment by Evengining** [comment]
@lloss Hi noustrum! Look if it's not hard for you, thanks!

**Pull request merged**